Change default garbage collection TTL of `od_url_metrics` posts from 1 month to 3 months and add the filter to customize expiration (#1950)
WordPress/performance: plugins/optimization-detective/docs/hooks.md

6. The list of active plugins.

A change in ETag means that any previously-collected URL Metrics will be immediately considered stale. When the ETag for URL Metrics in a complete viewport group no longer matches the current environment's ETag, new URL Metrics will then begin to be collected until there are no more stored URL Metrics with the old ETag.

> ### Filter: `od_url_metric_garbage_collection_ttl` (default: 3 * MONTH_IN_SECONDS)

Reviewer: ```suggestion
### Filter: `od_url_metric_garbage_collection_ttl` (default: 3 months in seconds)
```

Matches better with `od_url_metric_freshness_ttl`.

> Filters the expiration time (TTL) after which an unmodified od_url_metrics post not touched since that time will be garbage collected.

Reviewer: ```suggestion
Filters the expiration time (TTL) after which an unmodified `od_url_metrics` post not touched since that time will be garbage collected.
```

> Filters the expiration time (TTL) after which an unmodified `od_url_metrics` post not touched since that time will be garbage collected.

Reviewer: ```suggestion
Filters the expiration age (TTL) after which an `od_url_metrics` post will be garbage collected if it has not been modified since that time.
```

```php
add_filter( 'od_url_metric_garbage_collection_ttl', function (): int {
	return 6 * MONTH_IN_SECONDS;
} );
```

> To prevent garbage collection for od_url_metrics post, add the filter and return zero.

Reviewer: ```suggestion
To prevent garbage collection of `od_url_metrics` posts, add return zero from the filter:
```

> To prevent garbage collection of `od_url_metrics` posts, add return zero from the filter:

Reviewer: ```suggestion
To prevent garbage collection of `od_url_metrics` posts, add a filter that returns zero:
```
